adapters/bidmatic/bidmatic.go

```diff
@@ -0,0 +1,189 @@
+package bidmatic
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type bidmaticImpExt struct {
+	Bidmatic openrtb_ext.ExtImpBidmatic `json:"bidmatic"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	totalImps := len(request.Imp)
+	errors := make([]error, 0, totalImps)
+	imp2source := make(map[int][]int)
+
+	for i := 0; i < totalImps; i++ {
+
+		sourceId, err := validateImpression(&request.Imp[i])
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if _, ok := imp2source[sourceId]; !ok {
+			imp2source[sourceId] = make([]int, 0, totalImps-i)
+		}
+
+		imp2source[sourceId] = append(imp2source[sourceId], i)
```

Comment:
No need to check if  value doesn't exist for `imp2source[sourceId]`. Instead, could directly append value in `imp2source[sourceId]`

```
for i := 0; i < totalImps; i++ {
  sourceId, err := validateImpression(&request.Imp[i])
  if err != nil {
	errors = append(errors, err)
	continue
  }
 imp2source[sourceId] = append(imp2source[sourceId], i)
 

```

Reply:
We use it to optimize memory allocations